lib/semantics/resolve-names.cc

```diff
@@ -6272,14 +6273,30 @@ void ResolveNamesVisitor::ResolveExecutionParts(const ProgramTree &node) {
   SetScope(*node.scope());
   if (const auto *exec{node.exec()}) {
     Walk(*exec);
-    OmpAttributeVisitor{context(), *this}.Walk(*exec);
   }
   PopScope();  // converts unclassified entities into objects
   for (const auto &child : node.children()) {
     ResolveExecutionParts(child);
   }
 }
 
+// Resolve names for OpenMP in the specification and execution part of this node
+// and its children
+void ResolveNamesVisitor::ResolveOmpParts(const ProgramTree &node) {
+  if (!node.scope()) {
+    return;  // error occurred creating scope
+  }
+  SetScope(*node.scope());
+  OmpAttributeVisitor{context(), *this}.Walk(node.spec());
+  if (const auto *exec{node.exec()}) {
+    OmpAttributeVisitor{context(), *this}.Walk(*exec);
+  }
+  PopScope();  // converts unclassified entities into objects
```

Comment:
You don't need this, and the comment is inaccurate because that has already happened.